File: src/main/scala/com/github/dwendelen/platformd/rest/domain/budget/IncomeSource.java
package com.github.dwendelen.platformd.rest.domain.budget;

import com.datastax.driver.mapping.annotations.ClusteringColumn;
import com.datastax.driver.mapping.annotations.Column;
import com.datastax.driver.mapping.annotations.PartitionKey;
import com.datastax.driver.mapping.annotations.Table;

import java.util.UUID;

@Table(name = "income_source")
public class IncomeSource {
    @PartitionKey()
    @Column(name = "user_id")
    private UUID userId;
    @ClusteringColumn
    @Column(name = "income_source_id")
    private UUID uuid;
    @Column(name = "name")
    private String name;

    public UUID getUserId() {
        return userId;
    }

    public IncomeSource setUserId(UUID userId) {
        this.userId = userId;
        return this;
    }

    public UUID getUuid() {
        return uuid;
    }

    public IncomeSource setUuid(UUID uuid) {
        this.uuid = uuid;
        return this;
    }

    public String getName() {
        return name;
    }

    public IncomeSource setName(String name) {
        this.name = name;
        return this;
    }
}

File: application/src/main/scala/com/github/dwendelen/platformd/rest/domain/budget/IncomeSource.scala
package com.github.dwendelen.platformd.rest.domain.budget

import com.datastax.driver.mapping.annotations.{ClusteringColumn, Column, PartitionKey, Table}
import java.util.UUID

import scala.annotation.meta.field
import scala.beans.BeanProperty

@Table(name = "income_source") class IncomeSource (
    @(PartitionKey @field)
    @(Column @field)(name = "user_id")
    @BeanProperty
    var userId: UUID,
    @(ClusteringColumn @field)
    @(Column @field)(name = "income_source_id")
    @BeanProperty
    var uuid: UUID,
    @(Column @field)(name = "name")
    @BeanProperty
    var name: String
){
    def this() = this(null, null, null)
}